proposals/0000-accept-prs-on-github.rst

```diff
@@ -0,0 +1,89 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+
+.. highlight:: haskell
+
+Accept PRs on GitHub
+====================
+
+This proposes to accept pull requests (PRs) against GHC on GitHub (GH). There are a few
+possible mechanisms for doing so, as delineated below.
+
+This proposal comes directly out of a `conversation <https://mail.haskell.org/pipermail/ghc-devs/2016-September/012828.html>`_
+on the mailing lists.
+
+Motivation
+----------
+
+Lack of this feature to our workflow has been cited many times as a barrier to
+contributors. So let's do it!
+
+Proposed Change
+---------------
+
+* Start welcoming PRs on GH.
+
+From there, we have two choices:
+
+1. Automatically migrate all GH PRs to become Differentials in Phab. Users submitting
+   a PR on GH will be politely informed that their PR has been migrated to Phab, along
+   with instructions on how they can follow the PR.
+
+2. Create a tool (I will call it ``ghc-hub``) that GHC HQ (only) will use. This tool
+   allows migration of a GH PR to a Diff on Phab. GHC HQ will migrate "large" patches
+   (for an evolving definition of "large") to Phab and keep "small" patches on GH.
+   Generally, "small" patches will be accepted as-is; back-and-forth on a patch would
+   take place on Phab.
+
```

Comment:
> I think we have a third option here as well: politely ask the author to migrate his pr to phab, due to the size and the review that's needed, and provide a simple document with clear instructions how to do so.
This is likely a worthwhile option to mention, but I really don't think that it would be much work for a maintainer to push the PR to Phabricator. Moreover, it would be a small token to show the contributor "we care about your patch."

> This clearly puts the burden on the author, however can we expect the author to follow up on the review process on phabricator if this initial hurdle is already too high?

There is this question, but frankly if they don't follow-up then we can close it (say, after a standard two-weeks of inactivity). This seems perfectly reasonable to me. I honestly don't know how likely someone who is currently disengaged from GHC development would be to engage after having their patch moved to Phabricator. That being said, I suspect the answer is at least a bit more likely than they are currently.